components/org.wso2.carbon.identity.api.user.fido2/org.wso2.carbon.identity.api.user.fido2.common/src/main/java/org/wso2/carbon/identity/api/user/fido2/common/Util.java

```diff
@@ -0,0 +1,24 @@
+package org.wso2.carbon.identity.api.user.fido2.common;
+
+import org.wso2.carbon.identity.api.user.common.error.APIError;
+import org.wso2.carbon.identity.api.user.common.error.ErrorResponse;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Utility class for FIDO2 API.
+ */
+public class Util {
+
+    public static APIError handleError(Response.Status status, Constants.ErrorMessages error, String... data) {
+
+        String message;
+        if (data != null) {
+            message = String.format(error.getMessage(), data);
+        } else {
+            message = error.getMessage();
+        }
+        return new APIError(status, new ErrorResponse.Builder().withCode(error.getCode()).withMessage(message)
+                .withDescription(error.getDescription()).build());
```

Comment:
set the description with the above-construted instead of the message, as it is expected to provide details of the error why it went wrong.